Index: src/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Main.java b/src/MyLinkedListStack.java
rename from src/Main.java
rename to src/MyLinkedListStack.java
--- a/src/Main.java	(revision 9f2eea421e5562eaa46d5e4b2b13fa77758c8057)
+++ b/src/MyLinkedListStack.java	(date 1682865132725)
@@ -1,5 +1,36 @@
-public class Main {
-    public static void main(String[] args) {
-        System.out.println("Hello world!");
+import java.util.EmptyStackException;
+
+public class MyLinkedListStack<E> {
+
+    private MyLinkedList<E> list;
+
+    public MyLinkedListStack() {
+        list = new MyLinkedList<>();
     }
-}
\ No newline at end of file
+
+    public void push(E element) {
+        list.addFirst(element);
+    }
+
+    public E pop() {
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
+        return list.removeFirst();
+    }
+
+    public E peek() {
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
+        return list.getFirst();
+    }
+
+    public boolean isEmpty() {
+        return list.isEmpty();
+    }
+
+    public int size() {
+        return list.size();
+    }
+}
